daemon/create_windows.go

```diff
@@ -28,66 +28,76 @@ func (daemon *Daemon) createContainerOSSpecificSettings(container *container.Con
 		}
 		hostConfig.Isolation = "hyperv"
 	}
-	parser := volumemounts.NewParser(container.OS)
-	for spec := range config.Volumes {
-
-		mp, err := parser.ParseMountRaw(spec, hostConfig.VolumeDriver)
-		if err != nil {
-			return fmt.Errorf("Unrecognised volume spec: %v", err)
-		}
 
-		// If the mountpoint doesn't have a name, generate one.
-		if len(mp.Name) == 0 {
-			mp.Name = stringid.GenerateNonCryptoID()
```

Comment:
Bit confused by this one; I see that a name is now always generated; is this code only called for the Dockerfile `VOLUME` case, and not for (e.g.) the `-v volumename:C:/somepath` case?

Reply:
Ok, interestingly, the linux variant looks to do the same (still curious); in that case could you sync both implementations to copy the corresponding code, so that it's more clear where the differences are between the "windows" and "linux" implementation?

```patch
diff --git a/daemon/create_windows.go b/daemon/create_windows.go
index 7c15fc09c9..251acb83d1 100644
--- a/daemon/create_windows.go
+++ b/daemon/create_windows.go
@@ -30,6 +30,8 @@ func (daemon *Daemon) createContainerOSSpecificSettings(container *container.Con
        }
 
        for spec := range config.Volumes {
+               name := stringid.GenerateNonCryptoID()
+
                var destination string
                if container.OS == runtime.GOOS {
                        // We do a filepath.FromSlash here to not break "legacy" Windows
@@ -46,17 +48,11 @@ func (daemon *Daemon) createContainerOSSpecificSettings(container *container.Con
                        continue
                }
 
-               // Create the volume in the volume driver.
-               v, err := daemon.volumes.Create(
-                       context.TODO(),
-                       stringid.GenerateNonCryptoID(),
-                       hostConfig.VolumeDriver,
-                       volumeopts.WithCreateReference(container.ID))
+               v, err := daemon.volumes.Create(context.TODO(), name, hostConfig.VolumeDriver, volumeopts.WithCreateReference(container.ID))
                if err != nil {
                        return err
                }
 
-               // Add it to container.MountPoints. Note that the last parameter is true - as in read-write
                container.AddMountPointWithVolume(destination, &volumeWrapper{v: v, s: daemon.volumes}, true)
        }
```


One thing I notice being present in the Linux version, but not present on Windows is this; https://github.com/moby/moby/blob/b3e9f7b13b0f0c414fa6253e1f17a86b2cff68b5/daemon/create_unix.go#L52-L60

Would that also be needed on the Windows / LCOW implementation?